src/Common/hooks/useFilters.tsx

```diff
@@ -18,15 +19,25 @@ interface FilterBadgeProps {
  * A custom hook wrapped around raviger's `useQueryParams` hook to ease handling
  * of pagination and filters.
  */
-export default function useFilters({ limit = 14 }: { limit?: number }) {
+export default function useFilters({
+  limit = 14,
+  cacheBlacklist = [],
+}: {
+  limit?: number;
+  cacheBlacklist?: string[];
+}) {
   const { t } = useTranslation();
   const { kasp_string } = useConfig();
   const hasPagination = limit > 0;
   const [showFilters, setShowFilters] = useState(false);
   const [qParams, setQueryParams] = useQueryParams();
+  const [primaryCacheBlacklist, setPrimaryCacheBlacklist] =
+    useState(cacheBlacklist);
 
-  const updateQuery = (filter: FilterState) => {
+  const updateQuery = (filter: FilterState, cacheBlacklist: string[] = []) => {
     filter = hasPagination ? { page: 1, limit, ...filter } : filter;
+    if (cacheBlacklist.length)
+      setPrimaryCacheBlacklist(primaryCacheBlacklist.concat(cacheBlacklist));
```

Comment:
Why does `updateQuery` keep updating the cache blacklist? We can resolve it once and use that variable everywhere right?

```suggestion
  const updateQuery = (filter: FilterState) => {
    filter = hasPagination ? { page: 1, limit, ...filter } : filter;
```